contracts/poap/src/contract.rs

```diff
@@ -0,0 +1,1375 @@
+
+        let info = mock_info(MINTER, &vec![]);
+        let result = execute(deps.as_mut(), env.clone(), info, msg.clone());
+        // Minter can't update minter
+        assert_eq!(ContractError::Unauthorized {}, result.unwrap_err());
+
+        let info = mock_info(ADMIN, &vec![]);
+        execute(deps.as_mut(), env.clone(), info, msg.clone()).unwrap();
+
+        let config = CONFIG.load(&deps.storage).unwrap();
+        assert_eq!(NEW_MINTER, config.minter.as_str());
+    }
+
+    #[test]
+    fn mint_with_event_not_started_error() {
+        let mut deps = mock_dependencies();
+        let mut env = mock_env();
+        let info = mock_info(ADMIN, &vec![]);
+
+        do_instantiate(deps.as_mut());
+
+        env.block.time = Timestamp::from_seconds(INITIAL_BLOCK_TIME_SECONDS);
+
+        // Enable mint since is disable by default.
+        let msg = ExecuteMsg::EnableMint {};
+        execute(deps.as_mut(), env.clone(), info, msg).unwrap();
+
+        let msg = ExecuteMsg::Mint {};
+        let info = mock_info(USER, &vec![]);
+        let result = execute(deps.as_mut(), env.clone(), info, msg);
+
+        // Event is not started
+        assert_eq!(
+            ContractError::EventNotStarted {
+                current_time: env.block.time,
+                start_time: Timestamp::from_seconds(EVENT_START_SECONDS),
+            },
+            result.unwrap_err()
+        );
+    }
+
+    #[test]
+    fn mint_with_event_terminated_error() {
+        let mut deps = mock_dependencies();
+        let mut env = mock_env();
+        let info = mock_info(ADMIN, &vec![]);
+
+        do_instantiate(deps.as_mut());
+
+        // Enable mint since is disable by default.
+        let msg = ExecuteMsg::EnableMint {};
+        execute(deps.as_mut(), env.clone(), info, msg).unwrap();
+
+        env.block.time = Timestamp::from_seconds(EVENT_END_SECONDS);
+
+        let msg = ExecuteMsg::Mint {};
+        let info = mock_info(USER, &vec![]);
+        let result = execute(deps.as_mut(), env.clone(), info, msg);
+
+        // Event is not started
+        assert_eq!(
+            ContractError::EventTerminated {
+                current_time: env.block.time,
+                end_time: Timestamp::from_seconds(EVENT_END_SECONDS),
+            },
+            result.unwrap_err()
+        );
+    }
+
+    #[test]
+    fn mint_without_permissions_error() {
+        let mut deps = mock_dependencies();
+        let mut env = mock_env();
+        let info = mock_info(USER, &vec![]);
+
+        do_instantiate(deps.as_mut());
+
+        // Change current time to event start
+        env.block.time = Timestamp::from_seconds(EVENT_START_SECONDS);
+
+        let msg = ExecuteMsg::Mint {};
+        let result = execute(deps.as_mut(), env.clone(), info, msg);
+
+        // Event is not started
+        assert_eq!(ContractError::MintDisabled {}, result.unwrap_err());
+    }
+
+    #[test]
+    fn mint_limit() {
```

Comment:
It can be splitted into:
1. mint_token_properly
2. mint_out_of_max_amount_error

Reply:
The `mint_token_properly` is already coverd inside the integration tests.
This can just be renamed into `mint_out_of_max_amount_error`